frontend/packages/knative-plugin/src/utils/knatify-utils.ts

```diff
@@ -0,0 +1,150 @@
+    spec: { template },
+  } = obj;
+  const { metadata: templateMetadata, spec: templateSpec } = template || {};
+  const { image, ports, imagePullPolicy, env, resources } = templateSpec?.containers[0] || {};
+
+  const healthChecks = getHealthChecksData(obj, 0);
+  const { readinessProbe, livenessProbe } = getProbesData(healthChecks, Resources.KnativeService);
+
+  const newKnativeDeployResource: K8sResourceKind = {
+    kind: ServiceModel.kind,
+    apiVersion: `${ServiceModel.apiGroup}/${ServiceModel.apiVersion}`,
+    metadata: {
+      name,
+      namespace,
+      labels,
+      annotations,
+    },
+    spec: {
+      template: {
+        metadata: {
+          labels: templateMetadata?.labels ?? {},
+          annotations: templateMetadata?.annotations ?? {},
+        },
+        spec: {
+          containers: [
+            {
+              image,
+              ...(ports?.length > 0 && {
+                ports: [
+                  {
+                    containerPort: ports[0]?.containerPort,
+                  },
+                ],
+              }),
+              ...(imagePullPolicy && {
+                imagePullPolicy,
+              }),
+              ...(env?.length > 0 && { env }),
+              ...(resources && { resources }),
+              ...(readinessProbe && {
+                readinessProbe,
+              }),
+              ...(livenessProbe && {
+                livenessProbe,
+              }),
+            },
+          ],
+        },
+      },
+    },
+  };
+  return newKnativeDeployResource;
+};
+
+export const knatifyResources = async (
+  rawFormData: DeployImageFormData,
+): Promise<K8sResourceKind> => {
+  const formData = ensurePortExists(rawFormData);
+  const {
+    registry,
+    isi: { tag: imageStreamTag, image },
+  } = formData;
+
+  let imageStreamUrl: string = image?.dockerImageReference ?? '';
+  if (registry === RegistryType.External && !imageStreamUrl) {
+    const imageStreamResponse = await createOrUpdateImageStream(formData, false);
+    const imageStreamRepo = imageStreamResponse.status.dockerImageRepository;
+    imageStreamUrl = imageStreamTag ? `${imageStreamRepo}:${imageStreamTag}` : imageStreamRepo;
+  }
+
+  const knDeploymentResource = getKnativeServiceDepResource(formData, imageStreamUrl);
+
+  return k8sCreate(ServiceModel, knDeploymentResource);
+};
+
+const getCommonInitialValues = (
+  ksvcResourceData: K8sResourceKind,
+  name: string,
+  namespace: string,
+) => {
+  const appGroupName = _.get(ksvcResourceData, 'metadata.labels["app.kubernetes.io/part-of"]');
+  const commonInitialValues = {
+    name,
+    application: {
+      name: appGroupName || '',
+      selectedKey: appGroupName || UNASSIGNED_KEY,
+    },
+    project: {
+      name: namespace,
+    },
+    route: getKsvcRouteData(ksvcResourceData),
+    resources: Resources.KnativeService,
+    serverless: getServerlessData(ksvcResourceData),
+    pipeline: {
+      enabled: false,
+    },
```

Comment:
As part of this story, even if the D/DC has pipelines, we are not creating one in KSVC right?

Reply:
nope pipelines will handle as separate ticket, will update description as well with what all it covers.